File: Leetcode/14-Longest-Common-Prefix/14-Longest-Common-Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs[0].size(), m = strs.size();
        string result = "";

        for(int i = 0; i < n; i++) {
            for (int j = 1; j < m; j++) {
                if (strs[j][i] == '\0' || strs[j][i] != strs[0][i]) {
                    return result;
                }
            }

            result += strs[0][i];
        }

        return result;
    }
};

File: Leetcode/14-Longest-Common-Prefix/14-Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""

        for index in range(len(strs[0])):
            currentChar = strs[0][index]
            for s in strs:
                if index >= len(s) or s[index] != currentChar:
                    return prefix
            
            prefix += currentChar

        return prefix